packager/live_packager_test.cc

```diff
@@ -649,6 +666,63 @@ TEST_F(LivePackagerBaseTest, VerifyAes128WithDecryption) {
   }
 }
 
+// Test demonstrates decrypting fmp4 source with one set of keys and
+// re-encrypting using a difference encryption schema (AES-128) and different
+// set of keys. As validation the re-encryption segments are decrypted and
+// compared against decrypted segments.
+TEST_F(LivePackagerBaseTest, VerifyPrdDecryptReEncrypt) {
+  std::vector<uint8_t> init_segment_buffer =
+      ReadTestDataFile("encrypted/prd_data/init.mp4");
+  ASSERT_FALSE(init_segment_buffer.empty());
+
+  media::AesCbcDecryptor decryptor(media::kPkcs5Padding,
+                                   media::AesCryptor::kUseConstantIv);
+  ASSERT_TRUE(decryptor.InitializeWithIv(key_, iv_));
+
+  for (unsigned int i = 1; i <= 7; i++) {
+    std::string segment_num = absl::StrFormat("encrypted/prd_data/%05d.m4s", i);
+    std::vector<uint8_t> segment_buffer = ReadTestDataFile(segment_num);
+    ASSERT_FALSE(segment_buffer.empty());
+
+    SegmentData init_seg(init_segment_buffer.data(),
+                         init_segment_buffer.size());
+    SegmentData media_seg(segment_buffer.data(), segment_buffer.size());
+
+    LiveConfig live_config;
+    live_config.format = LiveConfig::OutputFormat::TS;
+    live_config.track_type = LiveConfig::TrackType::VIDEO;
+    live_config.protection_scheme = LiveConfig::EncryptionScheme::AES_128;
+    live_config.segment_number = i;
+    auto* decryption_keys = new (shaka::RawKeyParams);
+    decryption_keys->key_map[kDrmLabel0].key_id = HexStringToVector(kKeyIdHex);
+    decryption_keys->key_map[kDrmLabel0].key = HexStringToVector(kKeyHex);
+    decryption_keys->key_map[kDrmLabel0].iv = HexStringToVector(kIvHex);
+    decryption_keys->key_map[kDrmLabel1].key_id = HexStringToVector(kKeyId2Hex);
+    decryption_keys->key_map[kDrmLabel1].key = HexStringToVector(kKey2Hex);
+    decryption_keys->key_map[kDrmLabel1].iv = HexStringToVector(kIv2Hex);
+    live_config.decryption_keys = decryption_keys;
+
+    SetupLivePackagerConfig(live_config);
+
+    FullSegmentBuffer out;
+    ASSERT_EQ(Status::OK, live_packager_->Package(init_seg, media_seg, out));
```

Comment:
could you please double-check may this call change something during a packaging, e.g. timescale ?